### PR TITLE
feat(Dataset): generate upload url independetly

### PR DESCRIPTION
**Pull request by nazarfil**
move generation of file uploadUrl to a separate mutation
deprecate uploadUrl in DatasetVersionFile

### CODE REVIEW

File: hexa/datasets/schema/mutations.py [hunk 1]
```diff
@@ -178,6 +178,33 @@ def resolve_delete_dataset_share(_, info, **kwargs):
         return {"success": False, "errors": ["PERMISSION_DENIED"]}
 
 
+@mutations.field("generateUploadUrl")
+def resolve_generate_upload_url(_, info, **kwargs):
```
Comment (qgerome) [inline hexa/datasets/schema/mutations.py:182]: I'm pretty sure there is a permission that need to be checked here ("update" or something like that)

**Review by qgerome — CHANGES_REQUESTED** [review]
I think the part where you create the DatasetVersionFile after the upload is missing (in the tests and in the mutation resolver.

File: hexa/datasets/schema/types.py [hunk 2]
```diff
@@ -189,6 +194,16 @@ def resolve_version_permissions_delete(obj: DatasetVersion, info, **kwargs):
     )
 
 
+@dataset_version_file_result_object.field("uploadUrl")
+def resolve_upload_url(obj, info, **kwargs):
+    try:
+        upload_url = generate_upload_url(obj.uri, obj.content_type)
+        return upload_url
+    except BucketObjectAlreadyExists as exc:
```
Comment (qgerome) [inline hexa/datasets/schema/types.py:202]: Nice

File: hexa/datasets/schema/mutations.py (hunk 1 above)
Comment (qgerome) [reply hexa/datasets/schema/mutations.py:182]: @nazarfil Did you address this ?

File: hexa/datasets/schema/types.py [hunk 3]
```diff
@@ -6,13 +8,16 @@
 from hexa.workspaces.models import Workspace
 from hexa.workspaces.schema.types import workspace_object, workspace_permissions
 
+from ...files.basefs import BucketObjectAlreadyExists
```
Comment (qgerome) [inline hexa/datasets/schema/types.py:11]: Better to write `hexa.files.basefs`

Comment (nazarfil) [reply hexa/datasets/schema/types.py]: True!

**Pull request merged**